### ISSUE TITLE
_thisEventHandler and _thisScript 

### ISSUE THREAD

**Issue by WillWaywell** [LordGolias/sqf#8]
_thisEventHandler and _thisScript are variables privatised by the engine just like the _this variable.

_thisEventHandler is passed to all event handler code.
https://community.bistudio.com/wiki/addEventHandler

_thisScript is passed to code executed using spawn of execVM.
https://community.bistudio.com/wiki/spawn
https://community.bistudio.com/wiki/execVM

As such they need to be recognised by the analyser as already privatised in the current scope (just like _this).

**Comment by LordGolias** [comment]
To check that I understood, the code for a test case for this would be

```
    def test_thisScript(self):
        code = '[] spawn {x = _thisScript}'
        analyser = analyze(parse(code))
        self.assertEqual(len(analyser.exceptions), 0)
```

which tests that `_thisScript` is defined within a `spawn` code.

Can you suggest an example for the other case?

**Comment by PabstMirror** [comment]
I belive this is semi-related:

We sometimes use code that expects similar behavior.
example: https://github.com/CBATeam/CBA_A3/blob/master/addons/hashes/fnc_hashEachPair.sqf#L47
Code isn't passed anything, the values `_key` and `_value` are just expected to be used.

In ace we had a very basic private var detection script that would use this
```
//IGNORE_PRIVATE_WARNING ["_unit", "_weapon",.....
```
to skip warning about those privates.

I have not looked into how difficult something like that would be to implement.

**Comment by LordGolias** [comment]
@PabstMirror yep, that makes sense. If ACE3 is already using that convention, then maybe we can adopt it.

I would probably use a different convention (e.g. `//!`) to distinguish between simple comments and compiler comments, but we could just add an option to choose the token to list locals to ignore.

**Comment by LordGolias** [comment]
@PabstMirror, this was implemented in b1816333bf8d82bca8389f2d6c832b7c88370124. 

Essentially, codes below gives no errors:

```
//IGNORE_PRIVATE_WARNING ["_unit"];
_unit = 2;
```

```
//USES_VARIABLES ["_unit"];
_unit = 2;
```

**Comment by LordGolias** [comment]
I called this fixed in 2078153 since it solves the `_thisScript`. The event handler is a bit trickier, so I will create a separate issue for it.
